wasm2c/wasm-rt.h

```diff
@@ -326,6 +354,31 @@ typedef struct {
   bool is64;
 } wasm_rt_memory_t;
 
+#ifdef WASM_RT_C11_AVAILABLE
+/** A shared Memory object. */
+typedef struct {
+  /** The linear memory data, with a byte length of `size`. The memory is marked
+   * atomic as it is shared and may have to be accessed with different memory
+   * orders --- sequential when being accessed atomically, relaxed otherwise.
+   * Unfortunately, the C standard does not state what happens if there are
+   * overlaps in two memory accesses which have a memory order, e.g., an
+   * atomic32 being read from the same location an atomic64 is read. One way to
+   * prevent optimizations from assuming non-overlapping behavior as typically
+   * done in C is to mark the memory as volatile. Thus the memory is atomic and
+   * volatile.*/
+  _Atomic volatile uint8_t* data;
+  /** The current and maximum page count for this Memory object. If there is no
+   * maximum, `max_pages` is 0xffffffffu (i.e. UINT32_MAX). */
+  uint64_t pages, max_pages;
```

Comment:
Are there other places we declare more than once var in a single decl like this?    I would tend to avoid this, but maybe there is precedent in wabt?

Reply:
Yup, this is to be consistent with the style in the `wasm_rt_memory_t` definition just before this. I'm not opposed to avoiding this, but we can follow up in a separate PR since I have to modify `wasm_rt_memory_t` as well when we change this.

Reply:
Made a separate branch locally to fix this. Will follow up in a separate PR.